### PR TITLE
HARP-12700: Refactor Image caches

### PR DESCRIPTION
**Pull request by atomicsulfate**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/image/ImageCache.ts [hunk 1]
```diff
@@ -504,27 +247,26 @@ export class ImageCache {
...
-                this.cancelLoading(cacheItem.imageItem);
-                owner.imageRemoved(cacheItem.imageItem.url);
+            if (cacheItem.owners.length > 0) {
+                // Do not delete the item, it is still used.
+                return;
             }
-        } else {
-            this.m_images.delete(cacheItem.imageItem.url);
-            this.cancelLoading(cacheItem.imageItem);
         }
```
Comment (StefanDachwitz) [inline @here/harp-mapview/lib/image/ImageCache.ts:267]: The owners are no longer informed about removal of an image, owner.imageRemoved() is no longer being called. How do multiple MapViewImageCache instances (for shared caching of images for multiple MapViews) get the information that their images (same URL but different names) are no longer available in the cache?

Comment (StefanDachwitz) [reply @here/harp-mapview/lib/image/ImageCache.ts:267]: The owner is no longer optional, so I guess the images can no longer be removed until all owners have them removed individually. imageRemoved() is not required anymore.

File: @here/harp-mapview/lib/image/ImageCache.ts [hunk 2]
```diff
@@ -141,36 +106,13 @@ export class ImageCache {
...
-     * Remove an image from the cache.
+     * Remove an image from the cache..
      *
      * @param url - URL of the image.
-     * @param owner - Optional: Specify {@link ImageCacheOwner} removing the image.
+     * @param owner - Optional: Specify {@link any} removing the image.
      * @returns `true` if image has been removed.
      */
-    removeImage(url: string, owner?: ImageCacheOwner): boolean {
+    removeImage(url: string, owner?: any): boolean {
```
Comment (StefanDachwitz) [inline @here/harp-mapview/lib/image/ImageCache.ts:115]: An accidental removal without owner will remove it for all owners without their knowledge. I think owner is no longer optional.

File: @here/harp-mapview/lib/image/ImageCache.ts [hunk 3]
```diff
@@ -6,64 +6,37 @@
...
+    owners: any[] = [];
 
     /**
      * Instantiates `ImageCacheItem`.
      *
      * @param imageItem - The {@link ImageItem} referenced by the associated owners.
      * @param owner - An optional first owner referencing the {@link ImageItem}.
      */
-    constructor(public imageItem: ImageItem, owner?: ImageCacheOwner) {
+    constructor(public imageItem: ImageItem, owner?: any) {
```
Comment (StefanDachwitz) [inline @here/harp-mapview/lib/image/ImageCache.ts:30]: owner should no longer be optional.

File: @here/harp-mapview/lib/image/ImageCache.ts [hunk 4]
```diff
@@ -504,27 +247,26 @@ export class ImageCache {
...
+     * Remove the cacheItem from cache, unless the item is used by another owner, in that case the
+     * link to the owner is removed from the item, just like a reference count.
      *
      * @param cacheItem The cache item to be removed.
-     * @param owner - Optional: Specify {@link ImageCacheOwner} removing the image.
+     * @param owner - Optional: Specify which owner ({@link any}) removes the image.
+     * If no owner is specified, the cache item is removed even if it has owners.
      */
-    private unlinkCacheItem(cacheItem: ImageCacheItem, owner?: ImageCacheOwner) {
+    private unlinkCacheItem(cacheItem: ImageCacheItem, owner?: any) {
```
Comment (StefanDachwitz) [inline @here/harp-mapview/lib/image/ImageCache.ts:257]: owner should no longer be optional.

**Review by StefanDachwitz — CHANGES_REQUESTED** [review]

File: @here/harp-mapview/lib/image/ImageCache.ts (hunk 1 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/image/ImageCache.ts:267]: As we discussed, let's remove this callbacks as they shouldn't be needed if every owner is responsible for their own content.

File: @here/harp-mapview/lib/image/ImageCache.ts (hunk 2 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/image/ImageCache.ts:115]: will do.

File: @here/harp-mapview/lib/image/ImageCache.ts (hunk 3 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/image/ImageCache.ts:30]: will do.

File: @here/harp-mapview/lib/image/ImageCache.ts (hunk 4 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/image/ImageCache.ts:257]: will do.

**Review by StefanDachwitz — APPROVED** [review]

**Pull request merged**